kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/food2door/OrderProcessor.java

```diff
@@ -11,6 +15,17 @@ public OrderProcessor(InformationService informationService, OrderService orderS
     }
 
     public OrderDeliveryDto process(OrderRequest orderRequest) {
+        List<Shop> shops = orderRequest.getQuantityOfOrderedProducts().keySet().stream()
+                .map(Product::getShop)
+                .distinct()
+                .collect(Collectors.toList());
+        for (Shop shop : shops) {
+            Map<Product, Integer> shopProducts = orderRequest.quantityOfOrderedProducts.entrySet().stream()
+                    .filter(entry -> entry.getKey().getShop().equals(shop))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            shop.order(shopProducts);
+        }
+
         boolean isOrdered = orderService.order(orderRequest.getProvider());
```

Comment:
Klasa `OrderService` prawdopodobnie nie jest już do niczego potrzebna. Twój `boolean` isOrdered powinien być ustawiony przed pętlą `for`, z domyślną wartością (`true` lub `false`). Ewentualne niepowodzenie zamówienia u któregoś z dostawców powinno go modyfikować. Po przejściu calego for-a, powinnaś mieć status zamówienia jako całości (uwzględniając sukces / porażkę zamówienia cząstkowego u każdego z dostawców).